### DISCUSSION TITLE
Adding Wheels for SEP package

### DISCUSSION THREAD

**Discussion by mworion** [sep-developers/sep#88] [Ideas]
Hi a general idea:
For users installation is always a big hurdle for them. So I try to keep is as simple as possible. The ideal answer to that is to use packages which already supply wheels with precompiled binaries. 
What I learned so far (as I test the installation with actions on GitHub, for all python versions (3.7 - 3.9) and for all supported OS (MacOS, Ubuntu, Windows), the wheels were built on GitHub successful.
Michel

**Comment by mworion** [comment]
I worked some more on that topic and I'm able to provide a GitHub action which generates the wheels. The next step to do is to add these wheels to the package. Nothing more to do.
Michel

**Comment by kbarbary** [comment]
This would be great! Do you think you could create a PR with the github action?
Would uploading the wheels to PyPI be a separate step, or could this be added to the action?

**Comment by mworion** [comment]
Hi Kyle,

I think after some really long hours to learn building wheels and uploading them myself I made it. Right now testing in a forked setup and PyPITest. But it looks good. Hopefully I could provide a solution for it.

Some questions do arise:
- As maintainer you need to generate a token on PyPI and have to put it as a secret to GitHub. Otherwise no upload will happen.
- You need to think about the condition, when this workflow should run. I put some ideas into the script, but finally it's about the way of working in the project.

Do you have some feedback already ?
Michel

BTW: I would open the usage to python 3.9 as well. Are there any constraints about it ?

**Reply by kbarbary** [reply]
Thanks so much for figuring this out! I'll take a look tonight after work.

**Reply by kbarbary** [reply]
I published a release candidate to test the system: https://pypi.org/project/sep/1.1.0rc1/#files . Do you think you could test by installing `sep==1.1.0rc1`? If it looks good I'll re-push the v1.1.0 tag to build wheels for v1.1.0!

**Comment by mworion** [comment]
Hi Kyle,

so far so good. I used sep==1.1.0rc1 in my application and build it. The installation on windows 10, macOS bigsur and Ubuntu 18.04 went fine, in all cases I see sep installed. The logs show that no compile was necessary. And finally: I tested the functions based on sep package: worked in all cases.

I wonder if you could add the wheels on PyPI to an existing version, but you will find out. Please let me know if that worked adding it later.

From my side. there is a go for it !

Michel

BTW: If you would like to have a look to the app I build: [https://github.com/mworion/MountWizzard4](url). I use sep for HFD and flux calculation (showing background is a nice to have). Finally I would like to implement a quick autofocus on the imaging side.

**Comment by kbarbary** [comment]
I released a new version `v1.1.1` that is the same as `v1.1.0` but has built wheels. (There was probably a way to add wheels to v1.1.0, but this seemed easier.)

**Comment by mworion** [comment]
That was my guess. Thanks for the release, now I could move forward.
